File: mobile/lib/models/product.dart
class Product {
  final String id;
  final String name;
  final int price;
  final int? originalPrice;
  final double rating;
  final int points;
  final String category;
  final String? image;
  final bool isNew;
  final int? discount;
  final bool deliveryFree;
  final String? description;

  Product({
    required this.id,
    required this.name,
    required this.price,
    this.originalPrice,
    required this.rating,
    required this.points,
    required this.category,
    this.image,
    this.isNew = false,
    this.discount,
    this.deliveryFree = false,
    this.description,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id']?.toString() ?? json['id']?.toString() ?? '',
      name: json['name'] ?? '',
      price: json['price'] ?? 0,
      originalPrice: json['originalPrice'],
      rating: (json['rating'] ?? 0).toDouble(),
      points: json['points'] ?? 0,
      category: json['category'] ?? '',
      image: json['image'],
      isNew: json['isNew'] ?? false,
      discount: json['discount'],
      deliveryFree: json['deliveryFree'] ?? false,
      description: json['description'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'price': price,
      'originalPrice': originalPrice,
      'rating': rating,
      'points': points,
      'category': category,
      'image': image,
      'isNew': isNew,
      'discount': discount,
      'deliveryFree': deliveryFree,
      'description': description,
    };
  }
}

File: backend/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true,
        trim: true
    },
    price: {
        type: Number,
        required: true,
        min: 0
    },
    originalPrice: {
        type: Number,
        min: 0
    },
    rating: {
        type: Number,
        min: 0,
        max: 5,
        default: 0
    },
    points: {
        type: Number,
        default: 0
    },
    category: {
        type: String,
        required: true,
        enum: ['Maillets', 'Accessoires', 'Enfant']
    },
    image: {
        type: String,
        default: ''
    },
    isNew: {
        type: Boolean,
        default: false
    },
    discount: {
        type: Number,
        min: 0,
        max: 100
    },
    deliveryFree: {
        type: Boolean,
        default: false
    },
    stock: {
        type: Number,
        default: 0,
        min: 0
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Product', productSchema);